ID.me WebVerify SDK/IDmeWebVerify.m

```diff
@@ -51,118 +53,151 @@ + (IDmeWebVerify *)sharedInstance {
+- (void)getUserProfileWithResult:(IDmeVerifyWebVerifyResults _Nonnull)webVerificationResults; {
+    [self getAccessTokenWithScope:self.keychainData.scope
+                  forceRefreshing:NO
+                           result:^(NSDictionary * _Nullable userProfile, NSError * _Nullable error, NSString * _Nullable accessToken) {
+        NSString *requestString = [NSString stringWithFormat:IDME_WEB_VERIFY_GET_USER_PROFILE, self.keychainData.scope, accessToken];
+
+        requestString = [requestString stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
+        NSURL *requestURL = [NSURL URLWithString:requestString];
+        NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:requestURL];
+        [request setValue:@"IDmeWebVerify-SDK-iOS" forHTTPHeaderField:@"X-API-ORIGIN"];
+
+        NSURLSession *session = [NSURLSession sharedSession];
+        NSURLSessionDataTask *task = [session dataTaskWithRequest:request completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+            dispatch_async(dispatch_get_main_queue(), ^{
+                NSHTTPURLResponse* httpResponse = (NSHTTPURLResponse*)response;
+                NSUInteger statusCode = [httpResponse statusCode];
+                if ([data length]) {
+                    NSDictionary *results = (NSDictionary *)[NSJSONSerialization JSONObjectWithData:data options:NSJSONReadingAllowFragments error:nil];
+                    NSDictionary *userProfile = [self testResultsForNull:results];
+                    if (statusCode == 200) {
+                        webVerificationResults(userProfile, nil, nil);
+                    } else if (statusCode == 401) {
+                        //TODO: refresh token
+                    }
+                } else {
+                    NSError *modifiedError = [[NSError alloc] initWithDomain:IDME_WEB_VERIFY_ERROR_DOMAIN
+                                                                        code:IDmeWebVerifyErrorCodeVerificationDidFailToFetchUserProfile
+                                                                    userInfo:error.userInfo];
+                    webVerificationResults(nil, modifiedError, nil);
+                }
+
+                // Dismiss _webViewController and clear _webView cache
+                if (self.webNavigationController) {
+                    [self destroyWebNavigationController:self];
+                }
+            });
+            
+        }];
+        [task resume];
+    }];
+}
+
+-(void)logout {
+    [self.keychainData clean];
+}
+
+- (void)getAccessTokenWithScope:(NSString*)scope forceRefreshing:(BOOL)force result:(IDmeVerifyWebVerifyResults)callback{
+    if (force) {
+        // TODO: refresh token
+        callback(nil, nil, nil);
+        return;
+    } else if (![scope isEqualToString:self.keychainData.scope]) {
+        [self logout];
+        [self verifyUserInViewController:[self topMostController] scope:scope withTokenResult:callback];
```

Comment:
I don't think we should show the login form automatically. This function `getAccessToken` is intended to just get a fresh token to make an API call, for example. Remember that this could be done in a background thread where the developer maybe don't want or can't show some UI. In this case I would call the callback with an error. **This mostly must be defined in the issue discussion**